Nodejs/Product/TestAdapterImpl/TestFilesUpdateWatcher.cs

```diff
@@ -181,19 +183,19 @@ private void CheckDisposed()
 
         public void Dispose()
         {
-            if (Interlocked.CompareExchange(ref this.disposed, DISPOSED, NOT_DISPOSED) == NOT_DISPOSED)
+            if (Interlocked.CompareExchange(ref this.disposed, DISPOSED, NOT_DISPOSED) == NOT_DISPOSED
+                && this.fileWatcher.IsValueCreated)
             {
                 foreach (var cookie in this.watchedFiles.Values)
                 {
-                    this.fileWatcher.UnadviseFileChange(cookie);
+                    this.fileWatcher.Value.UnadviseFileChange(cookie);
                 }
                 foreach (var cookie in this.watchedFolders.Values)
                 {
-                    this.fileWatcher.UnadviseDirChange(cookie);
+                    this.fileWatcher.Value.UnadviseDirChange(cookie);
                 }
                 this.watchedFiles.Clear();
                 this.watchedFolders.Clear();
-                this.fileWatcher = null;
```

Comment:
Might be worth keeping this null assignment just so we know the object isn't being used.

Reply:
The Lazy can't be assigned to. We do have a DISPOSED variable and I'm checking for IsValueCreated before clearing things out to avoid creating the value just to clear it.